server/src/operators/dataset_operator.rs

```diff
@@ -294,6 +346,7 @@ pub async fn delete_chunks_in_dataset(
             chunk_metadata_columns::chunk_metadata
                 .filter(chunk_metadata_columns::dataset_id.eq(id))
                 .filter(chunk_metadata_columns::id.gt(last_offset_id))
+                .filter(chunk_metadata_columns::created_at.lt(deleted_at))
```

Comment:
What's the point of this?

Reply:
Actually nice touch. The point of this is to make sure that we don't delete data which has been added post the clear call. This keeps that data safe if you are writing and clearing at the same time.